packages/flutter_tools/lib/src/macos/macos_device.dart

```diff
@@ -0,0 +1,129 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+
+import '../application_package.dart';
+import '../base/io.dart';
+import '../base/platform.dart';
+import '../base/process_manager.dart';
+import '../build_info.dart';
+import '../desktop.dart';
+import '../device.dart';
+import 'macos_workflow.dart';
+
+/// A device which represents a desktop MacOS target.
+class MacOSDevice extends Device {
+  MacOSDevice() : super('MacOS');
+
+  @override
+  void clearLogs() {}
+
+  @override
+  DeviceLogReader getLogReader({ApplicationPackage app}) => NoOpDeviceLogReader('macos');
+
+  @override
+  Future<bool> installApp(ApplicationPackage app) {
+    throw UnimplementedError();
+  }
+
+  @override
+  Future<bool> isAppInstalled(ApplicationPackage app) {
+    throw UnimplementedError();
+  }
+
+  @override
+  Future<bool> isLatestBuildInstalled(ApplicationPackage app) {
+    throw UnimplementedError();
+  }
+
+  @override
+  Future<bool> get isLocalEmulator async => false;
+
+  @override
+  bool isSupported() => true;
+
+  @override
+  String get name => 'MacOS';
+
+  @override
+  DevicePortForwarder get portForwarder => const NoOpDevicePortForwarder();
+
+  @override
+  Future<String> get sdkNameAndVersion async {
+    final ProcessResult result = await processManager.run(<String>['sw_vers']);
+    if (result.exitCode != 0) {
```

Comment:
Is it expected that this can happen?  Should we not throw a tool exit?

Reply:
Ahh, excellent - problem solved